File: pset6/cash.py
# import stuff
import cs50
import math


def main():
    cents = round(dollars() * 100)
    quarters = 25
    dimes = 10
    nickels = 5
    pennies = 1
    coins = 0
    total_coins = 0
    current = cents

    while (quarters <= current):
        coins = math.floor(cents / quarters)
        current = cents % quarters
        total_coins = coins

    while (dimes <= current):
        coins = math.floor(current / dimes)
        current = current % dimes
        total_coins += coins

    while (nickels <= current):
        coins = math.floor(current / nickels)
        current = current % nickels
        total_coins += coins

    while (pennies <= current):
        coins = math.floor(current / pennies)
        current = current % pennies
        total_coins += coins

    print(math.floor(total_coins))
# define dollar conversion from change owed


def dollars():
    while True:
        dollars = cs50.get_float("Change owed: ")
        if (dollars > 0.00):
            break
    return dollars


main()

File: pset1/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

float dollars(void);

int main(void)
{
    // Converting dollars into cents
    int cents = round(dollars() * 100);
    int quarters = 25;
    int dimes = 10;
    int nickels = 5;
    int pennies = 1;
    int coins = 0;
    int tc = 0;
    int current = cents;
     
    while (quarters <= current)
    {
        coins = cents / quarters;
        current = cents % quarters;
        tc = coins;
    }
    while (dimes <= current)
    {
        coins = current / dimes;
        current = current % dimes;
        tc += coins;
    }
    while (nickels <= current)
    {
        coins = current / nickels;
        current = current % nickels;
        tc += coins;
    }
    while (pennies <= current)
    {
        coins = current / pennies; 
        current = current % pennies;
        tc += coins;
    }
    printf("%i\n", tc);
}

// Prompting user for change owed
float dollars(void)
{
    float dollars;
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars < 0.00);
    return (dollars);
}